lib/widgets/store.dart

```diff
@@ -246,3 +249,36 @@ class LoadingPage extends StatelessWidget {
     return const Center(child: CircularProgressIndicator());
   }
 }
+
+/// A [State] that uses the ambient [PerAccountStore].
+///
+/// The ambient [PerAccountStore] can be replaced in some circumstances,
+/// such as when an event queue expires. See [PerAccountStoreWidget.of].
+/// When that happens, stateful widgets should
+/// - remove listeners on the old [PerAccountStore], and
+/// - add listeners on the new one.
+/// Use this mixin, overriding [onNewStore], to do that concisely.
+// TODO(#185): Until #185, I think [PerAccountStoreWidget.of] never actually
+//   returns a different [PerAccountStore] from one call to the next.
+//   But it will, and when it does, we want our [StatefulWidgets] to handle it.
+mixin PerAccountStoreAwareStateMixin<T extends StatefulWidget> on State<T> {
+  PerAccountStore? _store;
+
+  /// Called in the first [State.didChangeDependencies] call, and afterward
+  /// when [PerAccountStoreWidget.of] returns a different store than last time.
+  ///
+  /// In this, remove any listeners on the old store
+  /// and add them on the new store.
+  void onNewStore();
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+
+    final storeNow = PerAccountStoreWidget.of(context);
+    if (_store != storeNow) {
+      onNewStore();
+    }
+    _store = storeNow;
```

Comment:
```suggestion
    if (_store != storeNow) {
      _store = storeNow;
      onNewStore();
    }
```

Since we're already going to the trouble of checking whether it changed, might as well set it only when it did.